h5pt.go

```diff
@@ -108,11 +119,27 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 // Struct values must only have exported fields, otherwise extractValues will
 // panic.
 func extractValues(ad *appendData, data interface{}) error {
+	if data, ok := data.(interface{ CMarshal() ([]byte, error) }); ok {
+		raw, err := data.CMarshal()
+		if err != nil {
+			return err
+		}
+
+		tmp := make([]byte, ad.offset+len(raw))
+		copy(tmp, ad.cBuffer)
+		tmp = ad.cBuffer
+		copy(ad.cBuffer[ad.offset:], raw)
+
+		return nil
+	}
+
 	rv := reflect.Indirect(reflect.ValueOf(data))
 	rt := rv.Type()
 
-	var dataPtr unsafe.Pointer
-	var dataSize C.size_t
+	// We will at most append 8 bytes at one time.
```

Comment:
This doesn't seem like it's always needed (and when it is, not necessarily this much, so we run ahead of ourselves). In the first case of the big switch, e.g., there's no addition to the buffer. Can we not just use `append` at the site of appending?

Reply:
Will be gone with @sbinet propsal.